logging/net_appender_test.go

```diff
@@ -103,23 +103,26 @@ func makeServerForRobotLogger(t *testing.T) serverForRobotLogger {
 	return serverForRobotLogger{robotService, config, rpcServer.Stop}
 }
 
-func TestNetLoggerBatchWrites(t *testing.T) {
+func TestNetLoggerSync(t *testing.T) {
 	server := makeServerForRobotLogger(t)
 	defer server.stop()
 
 	netAppender, err := NewNetAppender(server.cloudConfig)
 	test.That(t, err, test.ShouldBeNil)
 
+	// This test is testing the behavior of sync(), so the background worker shouldn't be running at the same time.
```

Comment:
I double checked that adding a sleep here didn't reintroduce some failure mode. 

If that sleep _were_ a problem, and we needed more control that the net appender (tests) never* creates the background syncer, we could introduce a different `NewNetAppenderWithoutSyncer` constructor.